hass-addon-sunsynk-dev/run.py

```diff
@@ -49,27 +49,58 @@ async def publish_sensors(sensors: List[Filter], *, force: bool = False) -> None
         )
 
 
-async def hass_discover_sensors(serial: str) -> None:
+async def hass_discover_sensors(serial: str, rated_power: float) -> None:
     """Discover all sensors."""
     ents: List[Entity] = []
     dev = Device(
         identifiers=[OPT.sunsynk_id],
         name=f"Sunsynk Inverter {serial}",
-        model=f"Inverter {serial}",
+        model=f"{int(rated_power/1000)}kW Inverter {serial}",
         manufacturer="Sunsynk",
     )
 
+    def create_on_change_handler(filt: Filter, value_func: Callable):
+        async def _handler(value):
+            await write_sensor(filt, value_func(value))
+
+        return _handler
+
+    async def static_or_sensor_value(val):
+        if isinstance(val, Sensor):
+            if val.value is None:
+                # Read sensor value for the first time
+                await read_sensors([val], retry_single=True)
+            return val.value
+        return val
+
     for filt in SENSORS:
         sensor = filt.sensor
-        ents.append(
-            SensorEntity(
-                name=f"{OPT.sensor_prefix} {sensor.name}".strip(),
-                state_topic=f"{SS_TOPIC}/{OPT.sunsynk_id}/{sensor.id}",
-                unit_of_measurement=sensor.unit,
-                unique_id=f"{OPT.sunsynk_id}_{sensor.id}",
-                device=dev,
+
+        if isinstance(sensor, NumberRWSensor):
+            ents.append(
+                NumberEntity(
+                    name=f"{OPT.sensor_prefix} {sensor.name}".strip(),
+                    entity_category="config",
+                    state_topic=f"{SS_TOPIC}/{OPT.sunsynk_id}/{sensor.id}",
+                    command_topic=f"{SS_TOPIC}/{OPT.sunsynk_id}/{sensor.id}_set",
+                    min=float(await static_or_sensor_value(sensor.min)),
+                    max=float(await static_or_sensor_value(sensor.max)),
+                    unit_of_measurement=sensor.unit,
+                    unique_id=f"{OPT.sunsynk_id}_{sensor.id}",
+                    device=dev,
+                    on_change=create_on_change_handler(filt, int),
+                )
+            )
+        else:
```

Comment:
Please use a continue here instead of the else. So by default it simply falls through